topic/machine-learning/llama-index/README.md

```diff
@@ -0,0 +1,118 @@
+# Connecting CrateDB Data to an LLM with LlamaIndex and Azure OpenAI
+
+This folder contains the codebase for [this tutorial](https://community.cratedb.com/t/how-to-connect-your-cratedb-data-to-llm-with-llamaindex-and-azure-openai/1612) on the CrateDB community forum.  You should read the tutorial for instructions on how to set up the components that you need on Azure, and use this README for setting up CrateDB and the Python code.
+
+This has been tested using:
+
+* Python 3.12.2
+* macOS Sequoia 15.0.1
+* CrateDB 5.8.3 running in CrateDB Cloud on AWS Europe (Ireland)
+
+## Database Setup
+
+You will need a CrateDB Cloud database: sign up [here](https://console.cratedb.cloud/) and use the free "CRFREE" tier.
+
+Make a note of the hostname, username and password for your database.  You'll need those when configuring the environment file later.
+
+Create a table in CrateDB:
+
+```sql
+CREATE TABLE IF NOT EXISTS time_series_data (
+    timestamp TIMESTAMP,
+    value DOUBLE,
+    location STRING,
+    sensor_id INT
+);
+```
+
+Add some sample data:
+
+```sql
+INSERT INTO time_series_data (timestamp, value, location, sensor_id)
+VALUES
+    ('2023-09-14T00:00:00', 10.5, 'Sensor A', 1),
+    ('2023-09-14T01:00:00', 15.2, 'Sensor A', 1),
+    ('2023-09-14T02:00:00', 18.9, 'Sensor A', 1),
+    ('2023-09-14T03:00:00', 12.7, 'Sensor B', 2),
+    ('2023-09-14T04:00:00', 17.3, 'Sensor B', 2),
+    ('2023-09-14T05:00:00', 20.1, 'Sensor B', 2),
+    ('2023-09-14T06:00:00', 22.5, 'Sensor A', 1),
+    ('2023-09-14T07:00:00', 18.3, 'Sensor A', 1),
+    ('2023-09-14T08:00:00', 16.8, 'Sensor A', 1),
+    ('2023-09-14T09:00:00', 14.6, 'Sensor B', 2),
+    ('2023-09-14T10:00:00', 13.2, 'Sensor B', 2),
+    ('2023-09-14T11:00:00', 11.7, 'Sensor B', 2);
+```
+
+## Python Project Setup
+
+Create and activate a virtual environment:
+
+```
+python -m venv venv
+. venv/bin/activate
```

Comment:
The `python` command ceased to exist, starting only recently. Let's also use a hidden directory, `.venv`.

```shell
$ python -V
pyenv: python: command not found
```

```suggestion
python3 -m venv .venv
source .venv/bin/activate
```